docs/general/architecture.md

```diff
@@ -0,0 +1,131 @@
+---
+sidebar_position: 3
+description: The inner workings of SLJIT.
+---
+
+# Architecture
+
+## Low-level Intermediate Representation
+
+Defining a LIR which provides wide range of optimization opportunities and still can be efficiently translated to machine code on all CPUs is the biggest challenge of this project.
+Those instruction forms and features which are supported on many (but not necessarily on all) architectures are carefully selected and a LIR is created from them.
+These features are also emulated by the remaining architectures with low overhead.
+For example, SLJIT supports various memory addressing modes and setting status register bits.
+
+## Generic CPU Model
+
+The CPU has:
+  - integer registers, which can store either an `int32_t` (4 byte) or `intptr_t` (4 or 8 byte) value
+  - floating point registers, which can store either a single (4 byte) or double (8 byte) precision value
+  - boolean status flags
```

Comment:
Some cpus have vector registers as well. On others vector registers also contain the floating point values.